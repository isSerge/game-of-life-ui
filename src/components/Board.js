import React from 'react'
import { v4 } from 'uuid'
import styled from 'styled-components'

import { TdBase } from './common'

const Td = styled(TdBase)`
    background-color: ${({ theme, color }) => (color ? color : theme.cellDefaultColor)};

    &:hover {
        background-color: ${({ theme, color }) => (color ? color : theme.hoverColor)};
    }
`

const Board = ({ cells, handleCellClick }) => (
    <table>
        <tbody>
            {cells.map((row, xIndex) => (
                <tr key={v4()}>
                    {row.map((cell, yIndex) => (
                        <Td
                            key={v4()}
                            color={cell}
                            onClick={() => handleCellClick(xIndex, yIndex)}
                        />
                    ))}
                </tr>
            ))}
        </tbody>
    </table>
)

export default Board
