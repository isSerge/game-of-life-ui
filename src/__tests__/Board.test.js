import React from 'react'
import Board from '../components/Board'
import renderer from 'react-test-renderer'

test('Board matches snapshot', () => {
    const props = {
        cells: [
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
        ],
        handleCellClick: jest.fn(),
    }
    const component = renderer.create(<Board {...props} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
