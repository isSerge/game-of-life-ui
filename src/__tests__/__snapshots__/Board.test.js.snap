// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board matches snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
    </tr>
    <tr>
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
    </tr>
    <tr>
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
    </tr>
    <tr>
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
    </tr>
    <tr>
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
      <td
        className="sc-bdVaJa sc-htpNat lndUsG"
        color={0}
        onClick={[Function]}
      />
    </tr>
  </tbody>
</table>
`;
